mpt/impl/imap-mailbox/core/src/main/resources/org/apache/james/imap/scripts/AuthenticatePlain.test

```diff
@@ -0,0 +1,56 @@
+################################################################
+# Licensed to the Apache Software Foundation (ASF) under one   #
+# or more contributor license agreements.  See the NOTICE file #
+# distributed with this work for additional information        #
+# regarding copyright ownership.  The ASF licenses this file   #
+# to you under the Apache License, Version 2.0 (the            #
+# "License"); you may not use this file except in compliance   #
+# with the License.  You may obtain a copy of the License at   #
+#                                                              #
+#   http://www.apache.org/licenses/LICENSE-2.0                 #
+#                                                              #
+# Unless required by applicable law or agreed to in writing,   #
+# software distributed under the License is distributed on an  #
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       #
+# KIND, either express or implied.  See the License for the    #
+# specific language governing permissions and limitations      #
+# under the License.                                           #
+################################################################
+
+# Correct user with correct password can authenticate with delegation part missing
+C: abcd AUTHENTICATE "PLAIN" {24+}
+# imapuser\0password
+C: aW1hcHVzZXIAcGFzc3dvcmQ=
+S: abcd OK AUTHENTICATE completed.
+
+REINIT
+
+# Correct user with correct password can authenticate with empty delegation part
+C: abcd AUTHENTICATE "PLAIN" {24+}
+# \0imapuser\0password
+C: AGltYXB1c2VyAHBhc3N3b3Jk
+S: abcd OK AUTHENTICATE completed.
+
+REINIT
+
+# Correct user with bad password cannot authenticate
+C: abcd AUTHENTICATE "PLAIN" {28+}
+# \0imapuser\0badpassword
+C: AGltYXB1c2VyAGJhZHBhc3N3b3Jk
+S: abcd NO AUTHENTICATE failed. Authentication failed.
+
+REINIT
+
+# Bad user cannot authenticate
+C: abcd AUTHENTICATE "PLAIN" {24+}
+# \0baduser\0password
+C: AGJhZHVzZXIAcGFzc3dvcmQ=
+S: abcd NO AUTHENTICATE failed. Authentication failed.
+
+REINIT
+
+# Correct user with correct password can authenticate with any delegation part
+C: abcd AUTHENTICATE "PLAIN" {36+}
+# delegate\0imapuser\0password
+C: ZGVsZWdhdGUAaW1hcHVzZXIAcGFzc3dvcmQ=
+S: abcd OK AUTHENTICATE completed.
```

Comment:
this test was written before your implementation, and it shows the current behaviour